### PR TITLE
setStyle endpoint

### LINKED ISSUE #16: API: create setStyle endpoint
### when a user completes a survey, update the user to have a `style` (request `operation` is `'setStyle'`)
  1. new endpoint to receive a user's style-survey result. request body will include `style`
  - possible backend entry file:  `/code/api/src/modules/user/resolvers`
  - frontend action location: `code/web/src/modules/user/api/actions` 

- [x] Create happy path for good requests
- [x] Add sad paths for bad requests
- [x] Add happy/sad path testing for setStyle

### PR DESCRIPTION
**Pull request by lkriffell**
Creates the setStyle endpoint for FE. It takes id and style (as a string) as args.
If the style is a blank string it throws an error
This pr also adds jest and 

Screen Shot 2020-12-07 at 3 43 26 PM

Example of what the graphQL response looks like.

To install the Sequelize CLI tools, from the root directory, navigate to code/api/ and in terminal type npm install --save-dev sequelize-cli. Afterwards, please run the following commands in order:

npx sequelize db:seed:undo:all
npx sequelize db:migrate
npx sequelize db:seed:all

From there you can npm start and run the query using GraphiQL

closes #16

### CODE REVIEW

**Comment by hashmaster3k** [comment]
@lkriffell the ID needed in the argument is the user's ID correct?

**Comment by lkriffell** [comment]
> @lkriffell the ID needed in the argument is the user's ID correct?

Yes

File: code/api/src/migrations/1-user.js [hunk 1]
```diff
@@ -29,6 +30,7 @@ module.exports = {
       }
     });
   },
+  // Provides a way to rollback if needed
```
Comment (hashmaster3k) [inline code/api/src/migrations/1-user.js:33]: Good idea on the comments to keep the code more approachable!

File: code/api/src/modules/tests/user/mutation.test.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    //
+    // it('delete an existing user', async (done) => {
+    //     const response = await request(server)
+    //         .post('/graphql')
+    //         .send({ query: 'mutation { userRemove(id:3) { id } }' })
+    //         .expect(200);
+    //         console.log(response.body);
+    //     done();
+    // });
+
```
Comment (hashmaster3k) [inline code/api/src/modules/tests/user/mutation.test.js]: Remove this commented out code?

File: code/api/src/modules/user/model.js [hunk 3]
```diff
@@ -1,5 +1,5 @@
 'use strict'
-
+// Model for user. Defines the types they are in the DB. Also defines associations between other resources such as Subscriptions
```
Comment (hashmaster3k) [inline code/api/src/modules/user/model.js]: Maybe we should remove these extra annotations?

File: code/api/src/modules/user/mutations.js [hunk 4]
```diff
@@ -37,4 +37,19 @@ export const userRemove = {
...
+      name: 'style',
+      type: GraphQLString
+    },
+    id: {
+      name: 'id',
+      type: GraphQLInt
+    }
+  },
+  resolve: updateStyle
+}
```
Comment (hashmaster3k) [inline code/api/src/modules/user/mutations.js:55]: Looks good!

File: code/api/src/modules/user/resolvers.js [hunk 5]
```diff
@@ -43,21 +43,35 @@ export async function login(parentValue, { email, password }) {
...
+      {
+        style
+      },
+      { where: { id } }
+    )
+  } else {
+    throw new Error(`You must enter a style to set a user's style`)
+  }
+}
+
```
Comment (hashmaster3k) [inline code/api/src/modules/user/resolvers.js:75]: This work around for now will allow for returning non-null response:

```node
export async function updateStyle(parentValue, { style, id }) {
  if (style !== "") {
    await models.User.update(
      {
        style
      },
      { where: { id } }
    );
    return await models.User.findOne({ where: { id } })
  } else {
    throw new Error(`You must enter a style to set a user's style`)
  }
}
```

File: code/api/src/modules/user/types.js [hunk 6]
```diff
@@ -6,12 +6,15 @@ const UserType = new GraphQLObjectType({
...
 
+
+  // defines a skeletal structure for return object and data types for user
   fields: () => ({
     id: { type: GraphQLInt },
     name: { type: GraphQLString },
     email: { type: GraphQLString },
     password: { type: GraphQLString },
     role: { type: GraphQLString },
+    style: { type: GraphQLString },
```
Comment (hashmaster3k) [inline code/api/src/modules/user/types.js:17]: Defined twice, see line 20 below. Please remove one or the other.

**Review by hashmaster3k — CHANGES_REQUESTED** [review]
Looks good overall, requests to make a few changes regarding extra annotations, setStyle function and an repeated type.

**Review by hashmaster3k — APPROVED** [review]

**Pull request merged**